### ISSUE TITLE
Feedback

### ISSUE THREAD

**Issue by donbowen** [LeDataSciFi/the-fin388-project-1-mba-ftw#1]
I was worried you didn't do the proposal - just put it under a diff name. Freaking MBAs, you're unbelievable. * shakes head * :)

Ok, so some reactions in the stream of consciousness:
- Very nice report. GOOD LAWD, a bibliography section?
- An "observation" seems to be a firm-day, and you'll have info on that firm's return on that day, along with variables from WSB from the day referring to that firm. 
- There are ways to include a reddit UN/PW combo without hard typing them into your code, I'd suggest you do that. 

My main constructive issues are that the analysis itself is weakly described. 
- You don't have a clear throughline connecting the research question to the hypo to the regression that will test that hypo. That hypothesis seems like it should be tested by regressing risk-adjusted returns (alphas) against your WSB variables and a volatility control. 
- So you'll need to obtain, for each firm, their factor loadings using 2020 data, then compute expected returns for each firm-day in your WSB sample period, and from that alpha
- I think a volatility control in the regression would be useful. High volatility days might drive "degenerates" to yolo, and such days might be have higher alphas for the types of firms WSB users trade on. 
- If returns is the independent variable, you wouldn't regress these on open/close/date.  Honestly I'm not sure what this gets at, except that you knew you'd have these variables lying around.

**Comment by donbowen** [comment]
Btw - You can submit your edits until Wednesday at 11pm.

**Comment by cak320** [comment]
Sorry about the folder mishap! Just one quick question: 

1. Why would we need to test by regressing risk-adjusted returns (alphas) against your WSB variables and a volatility control? Can we think of this as changes in stock returns?

**Comment by cak320** [comment]
Oh and another one: 

2. Is the volatility control based on WSB variable scores or stock return data? We think it's stock return data, but just want to make sure.

**Comment by donbowen** [comment]
You didn't submit a revision? :(

---

The basic idea is that high volatility for a stock could cause both (A) higher WSB activity - many users probably came to the subreddit because wild price movements got their attention and (B) high idiosyncratic (ie stock specific) volatility might drive increases in alpha for small stocks in a way that isn't going to be captured in a 3 factor model.

(A) + (B) = one way there might be a positive correlation between WSB and alpha that has nothing to do with WSB. So include a control for volatility. Let's define sigma(i,t) as the "estimated return volatility of firm i on day t" and measure it as the standard deviation of firm i's return over the past 20 days, meaning from day t-19 to day t (inclusive).

**Comment by donbowen** [comment]
> 
> 
> Sorry about the folder mishap! Just one quick question:
> 
>     1. Why would we need to test by regressing risk-adjusted returns (alphas) against your WSB variables and a volatility control? Can we think of this as changes in stock returns?

Well, you could regress raw returns on WSB variables. (And in fact, you should do this too! Most cross-sectional asset pricing studies report key regressions with raw returns and with alphas on the left hand side of the regression.) 

However, aggregate market factors might cause both returns to be high and WSB variables to be high. For example, if small firms are just broadly increasing in value (the size premium SMB is positive), that might also simultaneously cause WSB activity to spike (they mostly trade on small firms). 

So you "control" for that to isolate the WSB activity that is not due to general market conditions. And the "gold standard" is to estimate daily alphas for each firm and use those in plus of returns. 

So that would mean regress alpha(i,t) on WSBactivity(i,t-1). But I think vol is important too.

So your key table probably looks like:

1. return(i,t) = a+ B*WSBactivity(i,t-1) + u
2. return(i,t) = a+ B*WSBactivity(i,t-1) + C*[mkt, SMB, HML] + u    _(uses the 3 factor controls)_
2. return(i,t) = a+ B*WSBactivity(i,t-1) + C*[mkt, SMB, HML] + d*sigma(i,t) + u  
4. alpha(i,t) = a+ B*WSBactivity(i,t-1) + u        _(compare B from this to model 2)_
2. alpha(i,t) = a+ B*WSBactivity(i,t-1) + sigma(i,t) + u _(compare B from this to model 3)_

**Comment by donbowen** [comment]
Oh darnit, I remember, your proposal is in the ipynb file... just a minute. I'll look at that.

**Comment by donbowen** [comment]
Ok, you're ready to proceed. Good luck! A few final notes:

- Nitpicky on writing: "Final dataset will look like: Symbol (Tick), Date, Stock Return" is contradicted later. You'll have alpha, volatility, and WSB variable(s)
- I like the plan on WSB variables. You'll look for a ticker in a comment/post, and some indicator of buy/sell. 
    - Please test the regex functions you develop to make sure they work. For example, "seller's market" won't work if you clean the strings like in ASGN 5, because the apostrophe is replaced by a space.
    - Check for more sentiment words. Diamond hands? Rocket? Moon? Lambo?

**Comment by cak320** [comment]
Yes, we put your updates to the project proposal in the file. I think your clarifications make sense now but we'll get back with any questions ASAP. Thanks for your help, good call on the regex test, and more sentiment words :)
